FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.HealthCheker/Checkers/BaseHealthChecker.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Synapse.HealthCheker.Models;
+
+namespace Microsoft.Health.Fhir.Synapse.HealthCheker.Checkers
+{
+    public abstract class BaseHealthChecker : IHealthChecker
+    {
+        protected BaseHealthChecker(
+            string healthCheckName,
+            ILogger<BaseHealthChecker> logger)
+        {
+            Logger = EnsureArg.IsNotNull(logger, nameof(logger));
+            Name = EnsureArg.IsNotNullOrWhiteSpace(healthCheckName, nameof(healthCheckName));
+        }
+
+        protected ILogger<BaseHealthChecker> Logger { get; }
+
+        public string Name { get; set; }
+
+        public async Task<HealthCheckResult> PerformHealthCheckAsync(CancellationToken cancellationToken = default)
+        {
+            var healthCheckResult = new HealthCheckResult(Name);
+
+            try
+            {
+                await PerformHealthCheckImplAsync(healthCheckResult, cancellationToken);
```

Comment:
Return healthCheckResult directly in this function?

Reply:
> The health check mechanism needs to consider multiple scenarios, say we can take internal resources as the health check objects but not for external resources. We should have offline discussion with @tongwu-msft and @moria97 .

For this pr, we could first add external resource check, and refine it later.

Reply:
Agree. I